### PR TITLE
Add files for Notes feature to CA, SF and Data docs for edx.org

### PR DESCRIPTION
**Pull request by catong**
This PR is tracked by DOC-2732, DOC-2733 and TNL-2289
(https://openedx.atlassian.net/browse/DOC-2732, https://openedx.atlassian.net/browse/DOC-2733)

This PR re-adds the documentation from 2 reverted PRs (https://github.com/edx/edx-documentation/pull/809/ and https://github.com/edx/edx-documentation/pull/873) for the release of the Notes feature on edx.org.

The only edits made since the revert were to update the "Date Added" dates for the Notes events in tracking_logs.rst to 13 March 2016.
### Reviewers
- [x] Doc team review (sanity check/copy edit/dev edit): @srpearce or @lamagnifica or @pdesjardins 

FYI: @symbolist @muzaffaryousaf @scottrish 

FYI to reviewers. The Travis check is failing with known errors: there are cross-references from the Research Guide to the Notes topics in the course author and learner guides - these targets do not exist until those guides are built on RTD and will cause tests to fail.
### Testing
- [x] Ran ./run_tests.sh  - only known errors caused by xrefs to docs not yet built on RTD.
### Post-review
- [x] Add description to release notes task as a comment
- [x] Squash commits

### CODE REVIEW

**Comment by edx-webhook** [comment]
## [DOC-XXXX](https://openedx.atlassian.net/browse/DOC-XXXX)

Add a description of your changes.


### Date Needed (optional)

If the release date of a feature is known or estimated, provide it to give
reviewers guidance on turnaround time.


### Reviewers

Possible roles follow. PR submitter checks the boxes after each reviewer
finishes and gives :+1:. 

- [ ] Subject matter expert: 
- [ ] Subject matter expert: 
- [ ] Doc team review (sanity check/copy edit/dev edit): 
- [ ] Product review:
- [ ] Partner support: 
- [ ] PM review: 

FYI: Tag anyone else who might be interested in this PR here.


### Testing

- [ ] Ran ./run_tests.sh without warnings or errors


### HTML Version (optional)

- [ ] Build an RTD draft for your branch and add a link here

 
### Sandbox (optional)

- [ ] Point to or build a sandbox for the software change and add a link here


### Post-review

- [ ] Add description to release notes task as a comment
- [ ] Squash commits

**Comment by catong** [comment]
FYI to reviewers. The Travis check is failing with known errors: there are cross-references from the Research Guide to the Notes topics in the course author and learner guides - these targets do not exist until those guides are built on RTD and will cause tests to fail.

File: en_us/data/source/internal_data_formats/event_list.rst [hunk 1]
```diff
@@ -124,82 +138,6 @@ D, E, F
      - :ref:`problem`
    * - ``edx.problem.hint.feedback_displayed``
      - :ref:`problem`
-   * - ``edx.special_exam.practice.allowance.created``
```
Comment (pdesjardins) [inline en_us/data/source/internal_data_formats/event_list.rst:127]: It looks like this PR will revert changes I merged last night. Are there merge conflicts for this PR?

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 2]
```diff
@@ -2374,11 +2374,11 @@ field.
        * ``correct``: Boolean; 'true', 'false'
        * ``input_type``: string; The type of value that the student supplies
          for the ``response_type``. Based on the XML element names used in the
-         advanced editor. Examples include 'checkboxgroup', 'radiogroup',
+         Advanced Editor. Examples include 'checkboxgroup', 'radiogroup',
```
Comment (pdesjardins) [inline en_us/data/source/internal_data_formats/tracking_logs.rst]: I don't need to hold up the PR over capitalization, however we have been lower casing component names that are not edX trademarks (MS style). It looks like this change is reverting an earlier lower-casing. But we have a lot of clean up to do on that issue. I wouldn't worry about it here.

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 3]
```diff
@@ -2752,11 +2752,6 @@ The server emits ``showanswer`` events when the answer to a problem is shown.
          Studio, including the display name, problem text, and choices or response field
          labels.
 
-.. Including the special exam (timed and proctored) events doc in a separate
```
Comment (pdesjardins) [inline en_us/data/source/internal_data_formats/tracking_logs.rst:2755]: Here's another example of a change I merged last night.

File: en_us/data/source/internal_data_formats/tracking_logs.rst [hunk 4]
```diff
@@ -5888,11 +6125,5 @@ uploading a .csv file of student cohort assignments.
      - number
      - The numeric ID (from ``auth_user.id``) of the added user.
 
-.. Including the special exam (timed and proctored) developmnent events doc in
```
Comment (pdesjardins) [inline en_us/data/source/internal_data_formats/tracking_logs.rst:5891]: This is another part of my recent change.

File: en_us/shared/students/SFD_notes.rst [hunk 5]
```diff
@@ -109,43 +109,24 @@ You can view your course notes in two places.
...
-
-* When you select **Tags**, notes are sorted by the tags you have added.
-
-  * The most frequently used tags are at the top of the list. The number of
-    notes that have a particular tag appears in parentheses after the tag.
-  * If some tags have the same number of notes, those tags are sorted
-    alphabetically.
-  * Notes without tags are at the bottom of the list.
-
-  .. image:: ../../shared/students/Images/SFD_SN_TagsView.png
```
Comment (pdesjardins) [inline en_us/shared/students/SFD_notes.rst:144]: Should this image be removed from the repository since the reference to is gone?

**Comment by catong** [comment]
@pdesjardins Per our talk I'll make manual changes to restore your edits that you point out in this PR. I'll tag you again afterwards for verification.

**Comment by pdesjardins** [comment]
With the exception of the reverted changes for special exam events, :+1:. Also made some minor comments.

**Comment by catong** [comment]
@pdesjardins Please verify that special exams research guide content is restored. A draft project is here: http://draft-notesspecialexam-events.readthedocs.org/en/latest/internal_data_formats/tracking_logs.html#timed-and-proctored-exam-development-and-administration-events

File: en_us/shared/students/SFD_notes.rst (hunk 5 above)
Comment (srpearce) [inline en_us/shared/students/SFD_notes.rst:144]: This PR removes this file (just above the top of this file, SFD_SN_TagsView.png is listed with a "Deleted file not rendered" note), so we're good :-)

**Comment by srpearce** [comment]
:+1: LGTM

**Comment by pdesjardins** [comment]
@catong I verified that the special exam event doc is not being removed now. Looks good! :+1:

**Comment by catong** [comment]
Merging https://github.com/edx/edx-documentation/pull/924 with known reason for Travis failure.
**\* Not merging until feature is live on Prod.

**Comment by srpearce** [comment]
Just kidding! We found a problem with notes on Edge, so it's not being released this week after all.

**Comment by catong** [comment]
travis run tests

**Comment by catong** [comment]
run Travis

**Comment by catong** [comment]
Tests are failing on known issues relating to new links introduced in this PR. Merging https://github.com/edx/edx-documentation/pull/924.

**Pull request merged**